### PR TITLE
Make sure repo.url points to a git repo before adding

### PR DESCRIPTION
**Pull request by 00vareladavid**
What about something like this for #403?

### CODE REVIEW

**Comment by KristofferC** [comment]
Yes, this looks good! Thanks again.

The CI errors should be fixed after #456

File: src/GitTools.jl [hunk 1]
```diff
@@ -127,4 +127,34 @@ function fetch(repo::LibGit2.GitRepo, remoteurl=nothing; header=nothing, kwargs.
...
+                    err.code == LibGit2.Error.ENOTFOUND &&
+                    err.class == LibGit2.Error.Repository
+                Pkg.Types.cmderror("'$(url)' does not point to a valid git repository")
+            else
+                throw(err)
+            end
+        end
+    else
+        try
+            # try to fetch anything just to see if it throws an error
```
Comment (KristofferC) [inline src/GitTools.jl]: Won't the extra fetch give extra latency? Can't we put this inside `GitTools.fetch` and `GitTools.clone` to not do a redundant fetch?

Comment (00vareladavid) [reply src/GitTools.jl]: Ok, that would be a lot better.

**Comment by KristofferC** [comment]
Thanks, great!

**Comment by 00vareladavid** [comment]
Welcome!

**Pull request merged**